### PR TITLE
Float VMU option. Android renderer native thread. 90° screen rotation option. AW fixes. 

### PR DESCRIPTION
**Pull request by flyinghead**
- New option to display VMUs in game 

- Android: run renderer in a separate native thread 

- Android: add activity alias for intents backward-compatibility 

- AW: fix coin input. Subclass sega controller instead of #def 

- AW: setup maple controls depending on game 

- GL: Don't use screen stretching parameter for RTT 

- GL: 90deg screen rotation option. Fix scissor/clip wrt scale/stretch

### CODE REVIEW

File: core/hw/maple/maple_devs.cpp [hunk 1]
```diff
@@ -1299,6 +1329,10 @@ struct maple_mouse : maple_base
 
 struct maple_lightgun : maple_base
 {
+	virtual u32 process_kcode(u32 kcode) {
```
[the remark this answers is not in the document]
Comment (flyinghead) [reply core/hw/maple/maple_devs.cpp]: done

**Comment by skmp** [comment]
This pull request **introduces 2 alerts** and **fixes 4** when merging 13341ecd597ddb83cc31508d56c068b03f88a7cf into 56374377dd262f17272185d6c2292e5cefb5edcd - [view on LGTM.com](https://lgtm.com/projects/g/reicast/reicast-emulator/rev/pr-6070809bbcefe16274f86d80eeb678732f760842)

**new alerts:**

* 2 for FIXME comment

**fixed alerts:**

* 4 for FIXME comment

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: core/hw/aica/aica_if.cpp [hunk 2]
```diff
@@ -18,6 +18,7 @@ u32 VREG;//video reg =P
 u32 ARMRST;//arm reset reg
 u32 rtc_EN=0;
 int dma_sched_id;
+u32 RealTimeClock;
```
Comment (skmp) [inline core/hw/aica/aica_if.cpp]: In the future I think we should organize this thing in a struct called dc or such. Will make future refactoring easier

File: core/hw/maple/maple_cfg.cpp [hunk 3]
```diff
@@ -149,16 +149,27 @@ void mcfg_CreateAtomisWaveControllers()
 		// Game needs analog axes
 		mcfg_Create(MDT_SegaController, 2, 5, 0);
 		mcfg_Create(MDT_SegaController, 3, 5, 1);
+		// Faster Than Speed			needs 1 std controller on port 0 (digital inputs) and one on port 2 (analog axes)
+		// Maximum Speed				same
+	}
+	else if (settings.input.JammaSetup == 1)
```
Comment (skmp) [inline core/hw/maple/maple_cfg.cpp]: Prob. best to refer these by name instead of config ID

File: core/nullDC.cpp [hunk 4]
```diff
@@ -228,9 +230,11 @@ void LoadSpecialSettings()
 		printf("Enabling specific JVS setup for game %s\n", naomi_game_id);
 		settings.input.JammaSetup = 4;
 	}
```
Comment (skmp) [inline core/nullDC.cpp]: I think we need to move this logic to some built in gamedb in the future

File: core/rend/gl4/gles.cpp [hunk 5]
```diff
@@ -661,17 +679,41 @@ static bool RenderFrame()
 	/*
 		Handle Dc to screen scaling
 	*/
```
Comment (skmp) [inline core/rend/gl4/gles.cpp]: Does this mean RTTs are also scaled now?

File: core/types.h [hunk 6]
```diff
@@ -655,7 +657,6 @@ struct settings_t
 	struct
 	{
 		u32 cable;			// 0 -> VGA, 1 -> VGA, 2 -> RGB, 3 -> TV
-		u32 RTC;
```
Comment (skmp) [inline core/types.h]: Is there still some way to specify RTC eg for determinism or netplay?

File: core/rend/gl4/gles.cpp (hunk 5 above)
Comment (flyinghead) [reply core/rend/gl4/gles.cpp]: No, but they are rotated and they shouldn't. Good catch!

File: core/types.h (hunk 6 above)
Comment (flyinghead) [reply core/types.h]: not at the moment. It should probably be saved as part of the save state though.

**Pull request merged**